libr/search/aes-find.c

```diff
@@ -49,31 +49,53 @@ static bool aes128_key_test(const unsigned char *buf) {
 }
 
 R_API int r_search_aes_update(RSearch *s, ut64 from, const ut8 *buf, int len) {
-	int i, last = len - 20;
+	int i, t, last = len - 23;
 	RListIter *iter;
 	RSearchKeyword *kw;
+	const int old_nhits = s->nhits;
 
 	r_list_foreach (s->kws, iter, kw) {
 		if (last > 0) {
 			for (i = 0; i < last; i++) {
 				if (aes128_key_test (buf + i)) {
 					kw->keyword_length = 16;
-					return r_search_hit_new (s, kw, from + i);
+					t = r_search_hit_new (s, kw, from + i);
+					i += 24;
+					if (!t) {
+						return -1;
+					}
+					if (t > 1) {
+						return s->nhits - old_nhits;
+					}
 				}
-				if (len - i - 28 > 0) {
+				if (len - i - 31 > 0) {
```

Comment:
other hardcoded numbers

Reply:
I think you can do directly `if (len - i - SOME_CONSTANT > 0 && aes192_key_test (buf + i)) {`